criu/sk-unix.c

```diff
@@ -530,6 +530,12 @@ static int dump_one_unix_fd(int lfd, uint32_t id, const struct fd_parms *p)
 	if (dump_socket_opts(lfd, skopts))
 		goto err;
 
+	/* We need CAP_NET_ADMIN to restore SO_MARK, but we know the
+	 * service socket doesn't use it so avoid restoring it so that
+	 * the user doesn't have to grant the cap just for this. */
+	if (opts.unprivileged && (ue->uflags & USK_SERVICE))
+		skopts->has_so_mark = false;
```

Comment:
should we do something like this:
```
diff --git a/criu/sockets.c b/criu/sockets.c
index db772707b..43246bbf8 100644
--- a/criu/sockets.c
+++ b/criu/sockets.c
@@ -631,8 +631,9 @@ int dump_socket_opts(int sk, SkOptsEntry *soe)
        ret |= dump_opt(sk, SOL_SOCKET, SO_PRIORITY, &soe->so_priority);
        soe->has_so_rcvlowat = true;
        ret |= dump_opt(sk, SOL_SOCKET, SO_RCVLOWAT, &soe->so_rcvlowat);
-       soe->has_so_mark = true;
        ret |= dump_opt(sk, SOL_SOCKET, SO_MARK, &soe->so_mark);
+       if (soe->so_mark != 0)
+               soe->has_so_mark = true;
 
        ret |= dump_opt(sk, SOL_SOCKET, SO_SNDTIMEO, &tv);
        soe->so_snd_tmo_sec = tv.tv_sec;
```

Reply:
Thanks, I've changed the patch to do basically this but added `!unprivileged` so that existing behaviour for privileged mode is unchanged.